src/main/java/quotify_app/data_access/PredictionDataAccessObjectLiveTest.java

```diff
@@ -0,0 +1,202 @@
+                    throw new PredictionClientException("Simulated exception in predict()");
+                }
+            };
+
+            // Inject the mock PredictionClient into PredictionDataAccessObject
+            final PredictionDataAccessObject mockDataAccessObject = new PredictionDataAccessObject(mockClient);
+
+            // This should throw an exception
+            mockDataAccessObject.getFuturePricePredictions(property);
+        }
+        catch (PredictionClientException err) {
+            // Verify that the exception was correctly caught
+            Assertions.assertTrue(err.getMessage().contains("Simulated exception"),
+                    "Unexpected exception message: " + err.getMessage());
+            System.out.println("Caught expected exception: " + err.getMessage());
+        }
+    }
+
+    @Test
+    void testGetCurrentPricePredictionString() {
+        // Mock a successful current price prediction
+
+        final Property property = new Property(identifier, address, summary);
+
+        try {
+            // Call the live API to populate current price prediction
+            dataAccessObject.getCurrentPricePrediction(property);
+
+            // Fetch the current price prediction
+            final double currentPrice = dataAccessObject.getCurrentPricePredictionString();
+
+            // Assert that the returned value matches the expected range
+            Assertions.assertTrue(currentPrice > 0, "Current price prediction should be a positive value");
+            System.out.println("Retrieved Current Price Prediction: " + currentPrice);
+        }
+        catch (PredictionClientException err) {
+            Assertions.fail("Prediction failed: " + err.getMessage());
+        }
+    }
+
+    @Test
+    void testGetFuturePredictionsArray() {
+        // Mock a successful future price prediction
+
+        final Property property = new Property(identifier, address, summary);
+
+        try {
+            // Call the live API to populate future price predictions
+            dataAccessObject.getFuturePricePredictions(property);
+
+            // Fetch the future predictions array
+            final double[] futurePrices = dataAccessObject.getFuturePredictionsArray();
+
+            // Assert that the array has 6 elements, all positive
+            Assertions.assertEquals(numPredictions, futurePrices.length, "The predictions array should have 6 values");
+            for (double price : futurePrices) {
+                Assertions.assertTrue(price > 0, "Each future price prediction should be a positive value");
+            }
+
+            System.out.println("Retrieved Future Price Predictions: ");
+            for (double price : futurePrices) {
+                System.out.println(price);
+            }
+        }
+        catch (PredictionClientException err) {
+            Assertions.fail("Prediction failed: " + err.getMessage());
+        }
+    }
+
+}
+
```

Comment:
i am sort of leaving these for now, since u said the mockito was not working.

Reply:
On this file I included, everything was working -- i just didn't add tests that involved me hvaing to mock the httpclient used by the prediction client:

For example, I couldn't get coverage to 100% because one of the lines that I missed involved catching a jsonParsingError or something like that. In order to get that error to be thrown, I had to make an instance of the httpclient used in PredictionClient, but mockito doesn't support making mocks of HttpClient